pilot/pkg/bootstrap/server.go

```diff
@@ -1144,6 +1162,67 @@ func (s *Server) initControllers(args *PilotArgs) error {
 	return nil
 }
 
+func (s *Server) initFederationControllers(args *PilotArgs) error {
+	if !features.EnableFederation {
+		return nil
+	}
+
+	if s.kubeClient == nil {
+		log.Errorf("could not initialize federation discovery server: kubeClient is nil")
+		return nil
```

Comment:
Why aren't we failing here?

Reply:
I agree that we should remove error instead of fail silently and hide a problem. Btw., this line will not be reached if `kubeClient` was not initialized.

Reply:
Yeah I think we don't need that at all. I will remove it.

Reply:
Removed in 5a5daddfa9